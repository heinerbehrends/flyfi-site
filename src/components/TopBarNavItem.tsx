/** @jsx jsx */
import { jsx, Flex } from "theme-ui"
import React from "react"
import { Link } from "gatsby"

type TopBarNavItemProps = {
  text: string
  location: string
  variant?: "active"
}

export default function TopBarNavItem({
  text,
  location,
  variant,
}: TopBarNavItemProps) {
  return (
    <Link
      sx={{
        textDecoration: "none",
        color: `${variant === "active" ? "#fff" : "text"}`,
        fontSize: "3",
        fontWeight: variant === "active" ? 600 : 250,
        "&:focus": {
          outline: "none",
          boxShadow: "focus",
        },
      }}
      to={location}
    >
      <Flex
        sx={{
          height: "100%",
          backgroundColor: `${variant === "active" ? "muted" : "transparent"}`,
          "&:hover": { backgroundColor: "#c0eeee" },
          alignItems: "center",
          paddingX: "3",
        }}
      >
        {text}
      </Flex>
    </Link>
  )
}
