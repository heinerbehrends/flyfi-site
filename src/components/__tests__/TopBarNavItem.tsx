import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import TopBarNavItem from "../TopBarNavItem";

describe("TopBarNavItem", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<TopBarNavItem text="home" location="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders the active variant correctly", () => {
    const variantTree = renderer
      .create(<TopBarNavItem text="home" location="/" variant="active" />)
      .toJSON();
    expect(variantTree).toMatchSnapshot();
  });
  it("renders a link with correct label and location", () => {
    render(<TopBarNavItem text="home" location="/" />);
    expect(screen.getByText(/home/i)).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute("href", "/");
  });
});
