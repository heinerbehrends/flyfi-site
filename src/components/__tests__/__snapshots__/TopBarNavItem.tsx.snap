// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarNavItem renders correctly 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333;
  font-size: 16px;
  font-weight: thin;
  background-color: transparent;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 16px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-0:hover {
  background-color: #c0eeee;
}

.emotion-0:focus {
  outline: none;
  box-shadow: focus;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  padding-right: 8px;
}

@media screen and (min-width: 40em) {
  .emotion-1 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-1 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<a
  className="emotion-0"
  href="/"
>
  <div
    className="emotion-1"
  >
    home
  </div>
</a>
`;

exports[`TopBarNavItem renders the active variant correctly 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #333;
  font-size: 16px;
  font-weight: thin;
  background-color: transparent;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 16px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-0:hover {
  background-color: #c0eeee;
}

.emotion-0:focus {
  outline: none;
  box-shadow: focus;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 8px;
  padding-right: 8px;
}

@media screen and (min-width: 40em) {
  .emotion-1 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-1 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<a
  className="emotion-0"
  href="/"
>
  <div
    className="emotion-1"
  >
    home
  </div>
</a>
`;
